api/trace/b3_propagator.go

```diff
@@ -34,185 +34,295 @@ const (
 	b3TraceIDPadding = "0000000000000000"
 )
 
+var (
+	empty = EmptySpanContext()
+
+	errInvalidSampledByte        = errors.New("invalid B3 Sampled found")
+	errInvalidSampledHeader      = errors.New("invalid B3 Sampled header found")
+	errInvalidTraceIDHeader      = errors.New("invalid B3 TraceID header found")
+	errInvalidSpanIDHeader       = errors.New("invalid B3 SpanID header found")
+	errInvalidParentSpanIDHeader = errors.New("invalid B3 ParentSpanID header found")
+	errInvalidScope              = errors.New("require either both TraceID and SpanID or none")
+	errInvalidScopeParent        = errors.New("ParentSpanID requires both TraceID and SpanID to be available")
+	errInvalidScopeParentSingle  = errors.New("ParentSpanID requires TraceID, SpanID and Sampled to be available")
+	errEmptyContext              = errors.New("empty request context")
+	errInvalidTraceIDValue       = errors.New("invalid B3 TraceID value found")
+	errInvalidSpanIDValue        = errors.New("invalid B3 SpanID value found")
+	errInvalidParentSpanIDValue  = errors.New("invalid B3 ParentSpanID value found")
+)
+
+// B3Encoding is a bitmask representation of the B3 encoding type.
+type B3Encoding uint8
+
+const (
+	// MultipleHeader is a B3 encoding that uses multiple headers to
+	// transmit tracing information all prefixed with `x-b3-`.
+	MultipleHeader B3Encoding = 1
+	// SingleHeader is a B3 encoding that uses a single header named `b3 to
+	// transmit tracing information.
+	SingleHeader B3Encoding = 2
```

Comment:
```suggestion
	// B3MultipleHeader is a B3 encoding that uses multiple headers to
	// transmit tracing information all prefixed with `x-b3-`.
	B3MultipleHeader B3Encoding = 1
	// B3SingleHeader is a B3 encoding that uses a single header named `b3 to
	// transmit tracing information.
	B3SingleHeader B3Encoding = 2
```

These are exported from the `trace` package, so clarification of scope in the constant name might be useful.  I know the type also carries that information, but it is often not visible at the point of usage.

Reply:
This makes sense. The issue is going to be the fact that `B3SingleHeader` already is defined as the header key for the B3 context header.

I'm planning to just remove the exported header names (`B3SingleHeader`, `B3DebugFlagHeader`, `B3TraceIDHeader`, `B3SpanIDHeader`, `B3SampledHeader`, `B3ParentSpanIDHeader`) entirely. It seems like if the B3 header keys are needed [the authoritative OpenZipkin package constants](https://pkg.go.dev/github.com/openzipkin/zipkin-go@v0.2.2/propagation/b3?tab=doc#pkg-constants) should be used instead. We shouldn't be in the business of defining these.